apps/namadillo/src/App/Ibc/IbcWithdraw.tsx

```diff
@@ -17,110 +15,105 @@ import {
   namadaTransparentAssetsAtom,
 } from "atoms/balance";
 import { chainAtom } from "atoms/chain";
-import {
-  getChainRegistryByChainName,
-  ibcChannelsFamily,
-  namadaChainRegistryAtom,
-} from "atoms/integrations";
+import { ibcChannelsFamily } from "atoms/integrations";
 import { ledgerStatusDataAtom } from "atoms/ledger";
 import { createIbcTxAtom } from "atoms/transfer/atoms";
 import {
   clearDisposableSigner,
   persistDisposableSigner,
 } from "atoms/transfer/services";
 import BigNumber from "bignumber.js";
-import * as osmosis from "chain-registry/mainnet/osmosis";
 import { useFathomTracker } from "hooks/useFathomTracker";
-import { useRequiresNewShieldedSync } from "hooks/useRequiresNewShieldedSync";
 import { useTransaction } from "hooks/useTransaction";
 import { useTransactionActions } from "hooks/useTransactionActions";
-import { useUrlState } from "hooks/useUrlState";
 import { useWalletManager } from "hooks/useWalletManager";
-import { wallets } from "integrations";
 import { KeplrWalletManager } from "integrations/Keplr";
 import invariant from "invariant";
 import { useAtom, useAtomValue } from "jotai";
 import { TransactionPair } from "lib/query";
 import { useEffect, useState } from "react";
 import { generatePath, useNavigate } from "react-router-dom";
-import { Asset, IbcTransferTransactionData, TransferStep } from "types";
 import {
-  isNamadaAsset,
+  Asset,
+  AssetWithAmountAndChain,
+  IbcTransferTransactionData,
+  TransferStep,
+} from "types";
+import {
   toBaseAmount,
   toDisplayAmount,
   useTransactionEventListener,
 } from "utils";
-import { IbcTabNavigation } from "./IbcTabNavigation";
 import { IbcTopHeader } from "./IbcTopHeader";
 
-const defaultChainId = "cosmoshub-4";
-const keplr = new KeplrWalletManager();
-
-export const IbcWithdraw = (): JSX.Element => {
-  const defaultAccounts = useAtomValue(allDefaultAccountsAtom);
-  const shieldedAccount = defaultAccounts.data?.find(
-    (account) => account.type === AccountType.ShieldedKeys
-  );
-  const transparentAccount = useAtomValue(defaultAccountAtom);
-  const namadaChain = useAtomValue(chainAtom);
-  const [ledgerStatus, setLedgerStatusStop] = useAtom(ledgerStatusDataAtom);
-  const namadaChainRegistry = useAtomValue(namadaChainRegistryAtom);
-  const chain = namadaChainRegistry.data?.chain;
-
-  const requiresNewShieldedSync = useRequiresNewShieldedSync();
-  const [generalErrorMessage, setGeneralErrorMessage] = useState("");
-  const [selectedAssetAddress, setSelectedAssetAddress] = useUrlState(
-    params.asset
-  );
-  const [shielded, setShielded] = useState<boolean>(!requiresNewShieldedSync);
+interface IbcWithdrawProps {
+  sourceAddress: string | undefined;
```

Comment:
Is there a reason why here you use `| undefined` and few lines down just a `?`
like in `assetSelectorModalOpen?: boolean`